### PR TITLE
Converted the BaseEntity class to ES2015 Class syntax

### PR DESCRIPTION
**Pull request by spiray**
@MathuraMG @lm-n I decided to get rid of my previous work and start fresh. I wanted to start slow to make sure that we are fully on the same page. Here is 1 class (`BaseEntity`) converted to ES2015 syntax. Let me know what you think. We also need to make a decision on what we are replacing the parameter `arguments` with, for now i just changed it to `argumentss`. It is causing issues. Other than that keyword the examples are working fine with the updated class syntax.

### CODE REVIEW

**Comment by MathuraMG** [comment]
how about p5arguments?
Also, to clarify, is this (src/entities/_baseEntity.js) the only file with ES6 right now?

**Comment by MathuraMG** [comment]
The rules in the eslint look like a good start! Can we maybe add [no-var](https://eslint.org/docs/rules/no-var) too?

**Comment by spiray** [comment]
I started with that file. I will edit the script to execute on all js files in /src. I will add `no-var` as well.

**Comment by spiray** [comment]
I am getting an error when i test it. I can't pinpoint what is causing it though. 
image

File: examples/ExampleStyled/index.html [hunk 1]
```diff
@@ -1,22 +1,26 @@
 <html>
-  <head>
+
```
Comment (lm-n) [inline examples/ExampleStyled/index.html]: changes in this file look good and give no errors

File: package.json [hunk 2]
```diff
@@ -1,26 +1,36 @@
...
-    "grunt": "^0.4.5",
-    "grunt-contrib-concat": "^0.5.0",
-    "grunt-contrib-cssmin": "~0.10.0",
-    "grunt-contrib-sass": "^0.7.3",
-    "grunt-contrib-uglify": "^0.5.1",
-    "grunt-contrib-watch": "^0.6.1",
-    "grunt-contrib-coffee": "~0.12.0"
-  }
-}
+    "name": "p5-accessibility",
```
Comment (lm-n) [inline package.json]: changes in this file look good and give no errors

File: src/entities/_baseEntity.js [hunk 3]
```diff
@@ -1,96 +1,90 @@
...
-      }
-    } else {
-      if (yCoord < 0.4 * canvasY) {
-        return 'top middle';
-      } else if (yCoord > 0.6 * canvasY) {
-        return 'bottom middle';
-      } else {
-        return 'middle';
-      }
+class BaseEntity {
```
Comment (lm-n) [inline src/entities/_baseEntity.js]: This file throws errors

File: src/entities/backgroundEntity.js [hunk 4]
```diff
@@ -1,23 +1,19 @@
-function BackgroundEntity(Interceptor,object,arguments, canvasX, canvasY) {
-  var self = this;
-  var passedArguments = arguments;
-  this.populate = function(Interceptor) {
-    if(passedArguments[0].name == 'p5.Color') {
-      passedArguments = passedArguments[0].levels;
+class BackgroundEntity {
```
Comment (lm-n) [inline src/entities/backgroundEntity.js]: this file throws errors

File: src/entities/fillEntity.js [hunk 5]
```diff
@@ -1,23 +1,19 @@
-function FillEntity(Interceptor,shapeObject,arguments, canvasX, canvasY) {
-  var self = this;
-  var passedArguments = arguments;
-  this.populate = function(Interceptor) {
-    if(passedArguments[0].name == 'p5.Color') {
-      passedArguments = passedArguments[0].levels;
+class FillEntity {
+    constructor(Interceptor, shapeObject, argumentss, canvasX, canvasY) {
```
Comment (lm-n) [inline src/entities/fillEntity.js]: this file throws errors

File: src/entities/shapeEntity.js [hunk 6]
```diff
@@ -1,90 +1,90 @@
-function ShapeEntity(Interceptor,shapeObject,arguments, canvasX, canvasY) {
-  var self = this;
-  BaseEntity.call(self,shapeObject,arguments, canvasX, canvasY);
-  this.areaAbs = 0;
-  this.type = Interceptor.currentColor + ' ' + shapeObject.name;
-  this.area = 0;
+function ShapeEntity(Interceptor, shapeObject, arguments, canvasX, canvasY) {
```
Comment (lm-n) [inline src/entities/shapeEntity.js]: this file throws errors

File: src/entities/textEntity.js [hunk 7]
```diff
@@ -1,32 +1,25 @@
...
-  this.getAttributes = function() {
-    return({
-      type: this.type,
-      location: this.location,
-      coordinates: this.coordinates,
-    })
-  };
-
-  this.populate(shapeObject,arguments, canvasX, canvasY);
+class TextEntity {
```
Comment (lm-n) [inline src/entities/textEntity.js]: this file throws errors

File: src/textInterceptor/interceptorFunctions.js [hunk 8]
```diff
@@ -1,210 +1,211 @@
 function textInterceptor() { // eslint-disable-line
-  var self = this;
-  baseInterceptor.call(self);
+    let self = this;
```
Comment (lm-n) [inline src/textInterceptor/interceptorFunctions.js]: this file throws errors

**Comment by spiray** [comment]
@lm-n Yes, i will really take it slow this next time to make sure we are on the same page.

**Comment by lm-n** [comment]
@spiray the other files look good so we're happy to merge if you make the fix on src/textInterceptor/interceptorFunctions.js and bring the entities back to what they were.

**Comment by MathuraMG** [comment]
@spiray - totally agree with @lm-n, the work was good, it just needs the fixes :+1 
Also, we noticed that you are making changes on your master, if you instead make it on branches, it'll be easier to have multiple copies of the code!

**Pull request closed without merging**